Entity/BaseProduct.php

```diff
@@ -1112,4 +1118,48 @@ private function hasPriceInDefaultCurrency($prices, $defaultCurrency)
 
         return false;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTagNameArray()
```

Comment:
Also i'm not sure if this function should be part of the API entity. 
We try to use the entity as Data-Storage only (i know, we haven't been good at this here - until now)

to be discussed